server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/RouterSubsystemAdd.java

```diff
@@ -0,0 +1,106 @@
+      EndpointUtils.addSocketBindingDependency(builder, operation.get(ModelKeys.HOTROD_SOCKET_BINDING).asString(), routerService.getHotrodSocketBinding());
+      EndpointUtils.addSocketBindingDependency(builder, operation.get(ModelKeys.REST_SOCKET_BINDING).asString(), routerService.getRestSocketBinding());
+
+      ModelNode multiTenancyInnerConfiguration = config.get(ModelKeys.MULTI_TENANCY, ModelKeys.MULTI_TENANCY_NAME);
+
+      addHotRod(multiTenancyInnerConfiguration, routerService, builder);
+      addRest(multiTenancyInnerConfiguration, routerService, builder);
+
+      builder.install();
+   }
+
+   private void addRest(ModelNode config, MultiTenantRouterService routerService, ServiceBuilder<?> builder) {
+      if (config.get(ModelKeys.REST).isDefined()) {
+         config.get(ModelKeys.REST).asList().forEach(r -> {
+            String restName = r.get(0).get(ModelKeys.NAME).asString();
+            if (r.get(0).get(ModelKeys.PREFIX).isDefined()) {
+               r.get(0).get(ModelKeys.PREFIX).asList().forEach(prefix -> {
+                  String pathPrefix = prefix.get(0).get(ModelKeys.PATH).asString();
+                  MultiTenantRouterService.RestRouting restRouting = routerService.getRestRouting(pathPrefix, restName);
+                  EndpointUtils.addRestDependency(builder, restName, restRouting.getRest());
+               });
+            }
+         });
+      }
+   }
+
+   private void addHotRod(ModelNode config, MultiTenantRouterService routerService, ServiceBuilder<?> builder) {
+      if(config.get(ModelKeys.HOTROD).isDefined()) {
+         config.get(ModelKeys.HOTROD).asList().forEach(hr -> {
+            String hotRodName = hr.get(0).get(ModelKeys.NAME).asString();
+            routerService.tcpNoDelay(hr.get(0).get(ModelKeys.TCP_NODELAY).asBoolean(true));
+            routerService.keepAlive(hr.get(0).get(ModelKeys.KEEP_ALIVE).asBoolean(false));
+            routerService.sendBufferSize(hr.get(0).get(ModelKeys.SEND_BUFFER_SIZE).asInt(0));
+            routerService.receiveBufferSize(hr.get(0).get(ModelKeys.RECEIVE_BUFFER_SIZE).asInt(0));
+            if(hr.get(0).get(ModelKeys.SNI).isDefined()) {
+               hr.get(0).get(ModelKeys.SNI).asList().forEach(sni -> {
+                  String sniHostName = sni.get(0).get(ModelKeys.HOST_NAME).asString();
+                  String securityRealm = sni.get(0).get(ModelKeys.SECURITY_REALM).asString();
+                  MultiTenantRouterService.HotRodRouting hotRodRouting = routerService.getHotRodRouting(sniHostName);
+                  EndpointUtils.addHotRodDependency(builder, hotRodName, hotRodRouting.getHotRod());
+                  EndpointUtils.addSecurityRealmDependency(builder, securityRealm, hotRodRouting.getSecurityRealm());
+               });
+            }
+         });
+      }
+   }
+
+   private Optional<String> getServiceName(ModelNode config) {
+      return Optional.ofNullable(config.get(ModelKeys.NAME)).map(ModelNode::asString);
```

Comment:
`config.hasDefined(ModelKeys.NAME)` returns a `boolean`. If I used that I would have a lot of `if(config.hasDefined()){ return something }` constructs which I wanted to avoid.